File: go-backend/dto/audit.go
package dto

type ConnectionAuditUploadDto struct {
	Events []ConnectionAuditEventDto `json:"events"`
}

type ConnectionAuditEventDto struct {
	ServiceName string `json:"serviceName"`
	ClientAddr  string `json:"clientAddr"`
	ClientEmail string `json:"clientEmail"`
	TargetAddr  string `json:"targetAddr"`
	Protocol    string `json:"protocol"`
	UpBytes     int64  `json:"upBytes"`
	DownBytes   int64  `json:"downBytes"`
	DurationMs  int64  `json:"durationMs"`
	StartedTime int64  `json:"startedTime"`
	EndedTime   int64  `json:"endedTime"`
	Error       string `json:"error"`
}

type ConnectionAuditListDto struct {
	ClientIp    string `json:"clientIp"`
	ClientEmail string `json:"clientEmail"`
	Target      string `json:"target"`
	Service     string `json:"service"`
	ForwardId int64  `json:"forwardId"`
	NodeId    int64  `json:"nodeId"`
	NodeName  string `json:"nodeName"`
	Protocol  string `json:"protocol"`
	StartTime int64  `json:"startTime"`
	EndTime   int64  `json:"endTime"`
	Page      int    `json:"page"`
	PageSize  int    `json:"pageSize"`
}

File: nextjs-frontend/lib/api/audit.ts
import { post } from './client';

export type ConnectionAuditItem = {
  id: number;
  nodeId: number;
  nodeName?: string;
  serviceName?: string;
  forwardId: number;
  forwardName?: string;
  routeId: number;
  routeName?: string;
  tunnelId: number;
  tunnelName?: string;
  userId: number;
  userName?: string;
  clientAddr?: string;
  clientIp?: string;
  clientEmail?: string;
  clientPort: number;
  sourceType?: string;
  targetHost?: string;
  targetPort: number;
  targetAddr?: string;
  protocol?: string;
  upBytes: number;
  downBytes: number;
  durationMs: number;
  startedTime: number;
  endedTime: number;
  error?: string;
  createdTime: number;
};

export type ConnectionAuditListParams = {
  clientIp?: string;
  clientEmail?: string;
  target?: string;
  service?: string;
  forwardId?: number;
  nodeId?: number;
  nodeName?: string;
  protocol?: string;
  startTime?: number;
  endTime?: number;
  page?: number;
  pageSize?: number;
};

export type ConnectionAuditListResult = {
  list: ConnectionAuditItem[];
  total: number;
  page: number;
  pageSize: number;
};

export const getConnectionAuditList = (data: ConnectionAuditListParams) =>
  post<ConnectionAuditListResult>('/audit/list', data);